### PR TITLE
[PDI-13545] - Records are lost when you copy from a mapping to a step wi...

### PR DESCRIPTION
**Pull request by akhayrutdinov**
...th multiple copies

@mattyb149, @brosander review it please

### CODE REVIEW

**Comment by akhayrutdinov** [comment]
@mattyb149, @brosander, 
Just to remind you if you have missed it :)

File: engine/test-src/org/pentaho/di/trans/steps/mapping/MappingUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package org.pentaho.di.trans.steps.mapping;
```
Comment (mattyb149) [inline engine/test-src/org/pentaho/di/trans/steps/mapping/MappingUnitTest.java]: Needs a license/copyright header

**Comment by akhayrutdinov** [comment]
@mattyb149, have fixed it. Sorry, I forgot it again. Gonna change new file template in Intellij.
Look at this once more please.

File: test/org/pentaho/di/trans/steps/mapping/MappingTest.java [hunk 2]
```diff
@@ -250,4 +250,55 @@ public void testMapping_WhenSharingPreviousStepWithAnother() throws Exception {
...
+  /**
+   * This test case relates to PDI-13545. It executes a transformation with a Mapping step that is not configured
+   * manually with an <tt>outputStepname</tt> property and, therefore, it is set to be a mapping output step from the
+   * internal transformation.
+   *
+   * @throws Exception
+   */
+  public void testMapping_WhenNextStepHasTwoCopies_AndOutputIsNotDefinedExplicitly() throws Exception {
+    runTransWhenMappingsIsFollowedByCopiedStep(
+      "testfiles/org/pentaho/di/trans/steps/mapping/pdi-13545/PDI-13545-1.ktr" );
```
Comment (mattyb149) [inline test/org/pentaho/di/trans/steps/mapping/MappingTest.java]: The case of this filename does not match the actual filename (which is lowercase). It is causing a test failure.

File: test/org/pentaho/di/trans/steps/mapping/MappingTest.java [hunk 3]
```diff
@@ -250,4 +250,55 @@ public void testMapping_WhenSharingPreviousStepWithAnother() throws Exception {
...
+
+  /**
+   * This test case relates to PDI-13545. It executes a transformation with a Mapping step that is configured manually
+   * with an <tt>outputStepname</tt> property.
+   *
+   * @throws Exception
+   */
+  public void testMapping_WhenNextStepHasTwoCopies_AndOutputIsDefinedExplicitly() throws Exception {
+    runTransWhenMappingsIsFollowedByCopiedStep(
+      "testfiles/org/pentaho/di/trans/steps/mapping/pdi-13545/PDI-13545-2.ktr" );
```
Comment (mattyb149) [inline test/org/pentaho/di/trans/steps/mapping/MappingTest.java]: The case of this filename does not match the actual filename (which is lowercase). It is causing a test failure.

Comment (akhayrutdinov) [inline test/org/pentaho/di/trans/steps/mapping/MappingTest.java]: @mattyb149, have fixed it already: https://github.com/pentaho/pentaho-kettle/pull/1095/

**Pull request merged**